### PR TITLE
Fix null describedbys

### PR DESCRIPTION
**Pull request by DanCorder**
Fix aria-describedby values in various controls

### CODE REVIEW

File: GovUkDesignSystem/GovUkDesignSystemComponents/DateInput.cshtml [hunk 1]
```diff
@@ -2,9 +2,20 @@
...
+    string describedBy = Model.Fieldset?.DescribedBy;
+    string errorId = null;
+    string hintId = null;
+
+    if (Model.ErrorMessage != null)
+    {
+        errorId = Model.Id + "-error";
+        describedBy += $" {errorId}";
+    }
+    if(Model.Hint != null)
```
Comment (mattcottam91) [inline GovUkDesignSystem/GovUkDesignSystemComponents/DateInput.cshtml]: Tiny bit of missing whitespace :P

File: GovUkDesignSystem/GovUkDesignSystemComponents/DateInput.cshtml [hunk 2]
```diff
@@ -2,9 +2,20 @@
 @model GovUkDesignSystem.GovUkDesignSystemComponents.DateInputViewModel
 
 @{
-    var id = Model.Id;
-    var errorId = Model.ErrorMessage != null ? id + "-error" : null;
-    var hintId = Model.Hint != null ? id + "-hint" : null;
+    string describedBy = Model.Fieldset?.DescribedBy;
```
Comment (mattcottam91) [inline GovUkDesignSystem/GovUkDesignSystemComponents/DateInput.cshtml:5]: If I'm reading this and the way the tests are written correctly, we don't allow this to be null but the errorId or hintId to not be null. Is that right?

File: GovUkDesignSystem/GovUkDesignSystemComponents/ItemSet.cshtml [hunk 3]
```diff
@@ -4,8 +4,21 @@
...
+    string describedBy = Model.Fieldset?.DescribedBy;
+    string errorId = null;
+    string hintId = null;
+
+    if (Model.ErrorMessage != null)
+    {
+        errorId = idPrefix + "-error";
+        describedBy += $" {errorId}";
+    }
+    if(Model.Hint != null)
```
Comment (mattcottam91) [inline GovUkDesignSystem/GovUkDesignSystemComponents/ItemSet.cshtml]: Similar whitespace issue here

**Review by mattcottam91 — CHANGES_REQUESTED** [review]

File: GovUkDesignSystem/GovUkDesignSystemComponents/DateInput.cshtml (hunk 2 above)
Comment (DanCorder) [reply GovUkDesignSystem/GovUkDesignSystemComponents/DateInput.cshtml:5]: AFAIK any or all of them might be null (and this should still work)

Comment (mattcottam91) [reply GovUkDesignSystem/GovUkDesignSystemComponents/DateInput.cshtml:5]: Ah, you are right, looks like it converts nulls into empty strings during concatenation. Carry on then :)

File: GovUkDesignSystem/GovUkDesignSystemComponents/DateInput.cshtml (hunk 1 above)
Comment (DanCorder) [reply GovUkDesignSystem/GovUkDesignSystemComponents/DateInput.cshtml]: fixed

File: GovUkDesignSystem/GovUkDesignSystemComponents/ItemSet.cshtml (hunk 3 above)
Comment (DanCorder) [reply GovUkDesignSystem/GovUkDesignSystemComponents/ItemSet.cshtml]: fixed

**Pull request merged**